src/index.js

```diff
@@ -258,6 +258,7 @@ class Libp2p extends EventEmitter {
       await this.peerStore.stop()
       await this.connectionManager.stop()
 
+      ping.unmount(this)
```

Comment:
ping unhandle protocol is triggered, the identify-push will try to access the addresses when they do not exist

Reply:
This sounds like a bug. After the TransportManager is closed, we should have no active listeners, which should just yield an empty array of addresses.

This approach is also not good because we should not be doing an identify push when we are shutting down and this change is going to cause that to happen.

Reply:
If we do `await this.transportManager.close()` we will stop the transport listeners. With this, we have an error on `getAddrs()`, at list from `libp2p-tcp`: https://github.com/libp2p/js-libp2p-tcp/blob/master/src/listener.js#L96

So, when we do `ping.unmount`, it will call the `unhandle` which will try to do the `identify-push` and use the `libp2p.multiaddrs` getter, which uses the transport `getAddrs`, which fail as noted above. 

Or we unmount before, which in my opinion is correct as we are also unmounting the other protocols before, or we need to change the `transportManager.getAddrs()` to catch the transport.getAddrs error

Reply:
> This approach is also not good because we should not be doing an identify push when we are shutting down and this change is going to cause that to happen.

We are doing that for all other protocols. I think we should go with a more central fix for that. Before identify push, we should check that libp2p is not being stopped. What do you think?

Reply:
We have a check for `if (this.isStarted() && this.identifyService) {`. We can have a `._isStopping` property, or just turn false for isStarted before stopping the protocols

Reply:
> With this, we have an error on getAddrs(), at list from libp2p-tcp: https://github.com/libp2p/js-libp2p-tcp/blob/master/src/listener.js#L96

`close` sets all the listeners to empty, https://github.com/libp2p/js-libp2p/blob/v0.29.0/src/transport-manager.js#L72, and then `getAddrs` iterates over that list, which should immediately exit. The transport should **never** get called because it should already be cleaned up. I think there is a teardown race here.

Reply:
Ok, so I already understood the issue. In the test that causes this error we do:

```js
await libp2p.stop()
await libp2p.start()
```

When we do the libp2p stop, the `ping.unmount()` is not async. So the node will stop and going to restart while at the same time `ping.unmount` will call the unhandle while tcp server.listen has already kicked in from start. 

The best solution in my opinion is to add the `_isStopping` property. More than fixing this issue, it will block the identify-push from pubsub.stop, dht.stop from being triggered.

Reply:
For now, I just changed the `the._isStarted = false` for the beginning of stop. It seems not problematic to do this, as even if the stop code throws it would be assigned to false either way. We might get in a weird state if an error happens, but it would be problematic with the previous logic as well. A restart will fix the weird state